### DISCUSSION TITLE
Welcome to MineColab Discussions!

### DISCUSSION THREAD

**Discussion by thecoder-001** [thecoder-001/MineColab#13] [General]
## 👋 Welcome!
  Just a place to:
  * share some optimizations which might help others but are too case specific to be merged.
  * ask for help for smaller issues.
  * discussion for new ideas which haven't matured yet.
  * engage with other community members (if that ever happens).

## 📜 Remember:
  * Welcome others and be open-minded. 
  * Respect others' feelings.
  * Follow the [Github community guidelines](https://docs.github.com/en/free-pro-team@latest/github/site-policy/github-community-guidelines).

**Comment by broccoder** [comment]
Hi guys! I think we can talk about some stuff here now! Not sure what though.

**Comment by sqyo17** [comment]
So apparently there is a plugin that can modified java server to let bedrock version join the server. Thoughts?
https://geysermc.org/

**Reply by thecoder-001** [reply]
Geyser is quite nice and stable in my opinion. I have used it multiple times and have seen it used extensively on pretty large public servers. You would also need the plugin [floodgate](https://github.com/GeyserMC/Geyser/wiki/Floodgate), which would work with geyser to selectively bypass the java authentication for bedrock players. The only downsides are that java plugins that change game behavior would have some trouble or would not work at all for bedrock players and a bit of added latency.
Also, I was thinking, would it be a nice idea to include a script to setup some of commonly used plugins? That might be a little bit time saving to those who want to get things working fast.

**Reply by turtleship69** [reply]
You could have a number of wget scripts or something similar for the plug-ins that don't require setting up (most of them).

**Reply by jgx0** [reply]
the thing is, ngrok doesnt support UDP. Geyser just sets up UDP for the server

**Comment by turtleship69** [comment]
I ran the lower cell, then copied the first cell into a notebook I already had and always have open. If I need to update to the latest version of Minecraft do I need the lower cell. Also, is there a way to choose/install a specific version of Minecraft?

**Reply by thecoder-001** [reply]
Hello!
Yes you need the lower cell to update the server. Or you can just go to your google drive and navigate to `Minecraft-server`. There, delete the existing `server.jar` and upload your own one (don't forget to rename the new jar to `server.jar`!).
Also, yes, you can chose to install a specific version of the server by changing the value of `version` in the cell.

![Screen Shot 2020-12-25 at 10 52 37 AM](https://user-images.githubusercontent.com/25097841/103120815-51242600-469f-11eb-8345-c867dc23ea74.png)
Cheers

**Comment by broccoder** [comment]
We should probably start trying to make a colab side persistent config file soon (for compatibility with older versions of mc and plugins and some other stuff)
I still wonder how tho...
But me and @thecoder-001 both have lack of free time :disappointed:

**Comment by thecoder-001** [comment]
Hello everyone!
It would be epic if someone could work on merging any of the branches back to master.
Initial work has been done on the `config` branch which allows for storing the environmental variables but it has become outdated as compared to `master`.
For the `ngrok-mitigation` branch, while the original issue has vanished, it would still be a good idea to have multiple options for tunnelling service. Most of the work has already been done there, but the tunnel command execution part is partly broken in colab (seems to work fine locally) and a few bits need ironing out.
I still find it difficult to give much time because of my exams, but would hopefully be able to contribute more in the near future.

Cheers

**Reply by turtleship69** [reply]
Does lynk or tunnelto support UDP?
This would allow for both geysermc support, and (more usefully) bedrock.

**Reply by broccoder** [reply]
> Does lynk or tunnelto support UDP?
> This would allow for both geysermc support, and (more usefully) bedrock.

Unfortunately most likely not

**Reply by turtleship69** [reply]
:(

**Comment by turtleship69** [comment]
If anyone is good at working with google colabs forms, you could add a GUI and make certain things easier for example managing which minecraft version you are using and where you are saving it.

**Comment by pablodiaz2470** [comment]
Hello, your project has been the best I have seen and in a long time to be able to play servers among friends, I have two questions, 

1.What words do you accept in difficulty when establishing it?

2.How do I give op or admin on the server?, and I saw that there is a file that says ops.json but I don't understand what I should do

Please, I would like to know how to learn how to configure a server in the future, thank you
